core/src/wallet/bitcoin/api_impl/BitcoinLikeTransactionApi.cpp

```diff
@@ -268,6 +268,61 @@ namespace ledger {
             return writer.toByteArray();
         }
 
+        api::BitcoinLikeSignatureState BitcoinLikeTransactionApi::setSignatures(const std::vector<api::BitcoinLikeSignature> & signatures) {
+            if (signatures.size() != _inputs.size()) {
+                return api::BitcoinLikeSignatureState::MISSING_DATA;
+            }
+            for (std::size_t i = 0; i < signatures.size(); ++i) {
+                if (_inputs[i]->getScriptSig().size() > 0) {
+                    return api::BitcoinLikeSignatureState::ALREADY_SIGNED;
+                }
+                BytesWriter writer;
+                //Var bytes Signature (prefix length)
+                //Get length of VarInt representing length of R
+                BytesWriter rLength;
+                rLength.writeVarInt(signatures[i].r.size());
```

Comment:
Why do you use `writeVarInt`? In spec it should be just one byte, so the `writeByte` should be more appropriate.

Reply:
My bad, this is something I missed from the BIP.